tests/mainnet.rs

```diff
@@ -70,10 +71,20 @@ fn mainnet_blocks_from_rpc() {
 
             // TODO: Snapshot with consistent ordering for ease of diffing.
             // Currently [EvmStorage]'s storage ordering isn't consistent.
-            let accounts: BTreeMap<Address, EvmAccount> =
-                rpc_storage.get_cache_accounts().into_iter().collect();
+            let mut state = BTreeMap::<Address, EvmAccount>::new();
+            let mut bytecodes = BTreeMap::<B256, EvmCode>::new();
+            for (address, mut account) in rpc_storage.get_cache_accounts() {
+                if let Some(code) = account.code.take() {
+                    assert_ne!(account.code_hash.unwrap(), KECCAK_EMPTY);
+                    bytecodes.insert(account.code_hash.unwrap(), code);
+                }
+                state.insert(address, account);
+            }
+
             let file_state = File::create(format!("{dir}/pre_state.json")).unwrap();
-            serde_json::to_writer(file_state, &accounts).unwrap();
+            serde_json::to_writer(file_state, &state).unwrap();
+            let file_bytecodes = File::create(format!("{dir}/bytecodes.json")).unwrap();
```

Comment:
We should have a single `./data/bytecodes.json` that we read and update here instead of overwriting.